contribute/architecture/k8s-inspired-backend-arch.md

```diff
@@ -0,0 +1,288 @@
+# Grafana Kubernetes-inspired backend architecture
+
+Grafana is adopting a Kubernetes-inspired, resource-oriented model for all Grafana core resources (dashboards, folders, datasources, alert rules, ..) where users can interact with these resources by defining their desired state via well-structured objects sent to Grafana's new Resource APIs (`/apis/...`). Grafana then acts upon these declarations to align the system's current state with the requested state.
+
+On top of adopting the declarative, eventually-consistent system design, we are also adopting Kubernetes API's strict approach to URL structures, versioning and schemas. This approach addresses limitations of Grafana's traditional Legacy APIs (`/api/...`), including:
+- **Inconsistent designs:** Variations in structure, parameters, and response formats across different endpoints.
+- **No versioning:** Difficulty managing breaking changes and API evolution without clear versioning.
+- **Absence of schemas:** Lack of machine-readable schemas hindered programmatic interaction, validation, and tooling.
+- **Limited extensibility model:** No standardized way for Grafana plugins to extend core APIs.
+
+The end goal is for the Resource APIs to become the only interface for managing core Grafana resources. We will eventually deprecate and remove all Legacy API endpoints.
+
+## 1. Key Differences from Legacy API Endpoints
+
+- **URL structure & versioning:**
+  - **Resource APIs:** Follow Kubernetes conventions (`/apis/<group>/<version>/namespaces/<namespace>/<resource>/<name>`). Includes explicit API versions (e.g., `v0alpha1`, `v1alpha1`, `v2alpha1`) in the path, allowing for controlled evolution and multiple versions of a resource API to co-exist.
+  - **Legacy APIs:** Variable path structures (e.g., `/api/dashboards/uid/:uid`, `/api/ruler/grafana/api/v1/rules/:uid/:uid`). Less explicit versioning.
+
+- **Resource schemas:**
+  - **Resource APIs:** Each resource has a well-defined schema (`spec`) and is wrapped with an envelope with common metadata, all APIs come with an always-in-sync OpenAPI spec.
+  - **Legacy APIs:** Structures vary
+
+- **Namespacing / org context:**
+  - **Resource APIs:** Uses explicit namespaces in the path (`/namespaces/<namespace>/...`) mapping to Grafana Organization IDs for scoping.
```

Comment:
Org IDs or Instance IDs in cloud

Reply:
Does this means organizations will have uids or will those ids bé the currently séquence ids?

Reply:
@QuentinBisson - we will continue with orgIds for now, see https://github.com/grafana/authlib/blob/13c42d09f9f5cc06f3a64d36f75a024e2d93d229/types/namespace.go#L58

Reply:
👍🏻